Modules/GamePasses.lua

```diff
@@ -12,30 +12,37 @@ local boughtGamePasses = WeakInstanceTable()
 
 local listeningPasses = {}
 
-MarketplaceService.PromptGamePassPurchaseFinished:connect(function(player, gamePassId, purchased)
+MarketplaceService.PromptGamePassPurchaseFinished:Connect(function(player, gamePassId, purchased)
 	if not purchased then return end
 	boughtGamePasses[player] = boughtGamePasses[player] or {}
 	boughtGamePasses[player][gamePassId] = true
 end)
 
 local GamePasses = {}
 
+local function userOwnsGamePassAsync(player, gamePassId)
+	local success, doesOwn = pcall(MarketplaceService.UserOwnsGamePassAsync, MarketplaceService, player.UserId, gamePassId)
+	return success and doesOwn or false
+end
+
 function GamePasses.ListenForPass(gamePassId)
 	if listeningPasses[gamePassId] then return end
 	listeningPasses[gamePassId] = true
 
 	local function checkGamePassOwnership(player)
 		boughtGamePasses[player] = boughtGamePasses[player] or {}
-		boughtGamePasses[player][gamePassId] = MarketplaceService:UserOwnsGamePassAsync(player.UserId, gamePassId)
+		boughtGamePasses[player][gamePassId] = userOwnsGamePassAsync(player, gamePassId)
 	end
 
 	if RunService:IsServer() then
-		Players.PlayerAdded:connect(checkGamePassOwnership)
-		for _, player in pairs(Players:GetPlayers()) do
-			coroutine.wrap(checkGamePassOwnership)(player)
+		Players.PlayerAdded:Connect(checkGamePassOwnership)
+		for _, player in ipairs(Players:GetPlayers()) do
+			local thread = coroutine.create(checkGamePassOwnership)
+			coroutine.resume(thread, player)

```

Comment:
Why did you change this from wrap? It's more readable IMO.

Reply:
It is more readable, I probably changed it for either the fact that `coroutine.resume` returns success and error like `pcall` and it's a bit faster.

Reply:
We're not using the success/error here, and it would probably not work well here anyway since UserOwnsGamePassAsync yields